tests/components/mqtt/test_cover.py

```diff
@@ -224,18 +218,16 @@
     })
 
     state = hass.states.get('cover.test')
-    assert STATE_UNKNOWN == state.state
+    assert state.state == STATE_UNKNOWN
 
-    hass.async_add_job(
-        hass.services.async_call(
-            cover.DOMAIN, SERVICE_CLOSE_COVER,
-            {ATTR_ENTITY_ID: 'cover.test'}))
+    await hass.services.async_call(
+        cover.DOMAIN, SERVICE_CLOSE_COVER, {ATTR_ENTITY_ID: 'cover.test'})
     await hass.async_block_till_done()
```

Comment:
I think we can remove this if the service call is blocking and we await it.

Reply:
Yes, that's right, will fix!

Btw, the same pattern (add_job/create_task + async_block_till_done) is used in other MQTT tests via a `common.py` test helper, for example here:
https://github.com/home-assistant/home-assistant/blob/264dc898838c7f6d44801b2ce30b3f993ff44f5b/tests/components/mqtt/test_alarm_control_panel.py#L95-L96

https://github.com/home-assistant/home-assistant/blob/264dc898838c7f6d44801b2ce30b3f993ff44f5b/tests/components/alarm_control_panel/common.py#L55-L64

Should that also be rewritten as:
```diff
-    common.async_alarm_arm_home(hass)
-    await hass.async_block_till_done()
+    await hass.services.async_call(alarm.DOMAIN, SERVICE_ALARM_ARM_HOME, blocking=True)
```

Or should the `common.py` test helper be rewritten?

Reply:
The test helper isn't a coroutine so we can't await it. If we can change it to be a coroutine that is probably nicer. I'm not sure about the implications of that though.

Reply:
If you don't think we need the helper we can remove it too. :+1:

Reply:
It's mostly fine to make the helpers coroutines, but there are some negative tests such as this one which will fail since the test case will now fail if exceptions are thrown from the coroutine:
https://github.com/home-assistant/home-assistant/blob/a89c7f8feb6ac921793972900f652a3f9748d9e3/tests/components/mqtt/test_climate.py#L228-L239

Maybe such test cases can still use hass.async_create_task() to run the coroutine? or should they be rewritten to incude a `try` / `except` in the test case itself?

Reply:
We can use `with pytest.raises:`, I think.